src/hivpy/hiv_testing.py

```diff
@@ -84,29 +90,87 @@ def update_hiv_testing(self, pop):
+            r = rng.uniform(size=len(sub_pop))
+            tested = r < prob_test
+            pop.set_present_variable(col.LAST_TEST_DATE, pop.date,
+                                     sub_pop=pop.apply_bool_mask(tested, sub_pop))
+
+    def update_anc_hiv_testing(self, pop, time_step):
+        """
+        Update which pregnant women are tested while in antenatal care.
+        COVID disruption is factored in.
+        """
+        # conduct up to three tests in anc during pregnancy if there is no covid disruption
+        if not (self.covid_disrup_affected | self.testing_disrup_covid):
+            # get population at the end of the first trimester
+            first_trimester_pop = pop.get_sub_pop([(col.HIV_STATUS, op.eq, False),
+                                                   (col.ANC, op.eq, True),
+                                                   (col.LAST_PREGNANCY_DATE, op.le, pop.date
+                                                    - timedelta(days=90)),
+                                                   (col.LAST_PREGNANCY_DATE, op.gt, pop.date
+                                                    - (timedelta(days=90) + time_step))])
+            self.update_sub_pop_test_date(pop, first_trimester_pop, self.prob_anc_test_trim1)
+
+            # get population at the end of the second trimester
+            second_trimester_pop = pop.get_sub_pop([(col.HIV_STATUS, op.eq, False),
+                                                    (col.ANC, op.eq, True),
+                                                    (col.LAST_PREGNANCY_DATE, op.le, pop.date
+                                                     - timedelta(days=180)),
+                                                    (col.LAST_PREGNANCY_DATE, op.gt, pop.date
+                                                     - (timedelta(days=180) + time_step))])
+            self.update_sub_pop_test_date(pop, second_trimester_pop, self.prob_anc_test_trim2)
+
+            # get population at the end of the third trimester
+            third_trimester_pop = pop.get_sub_pop([(col.HIV_STATUS, op.eq, False),
+                                                   (col.ANC, op.eq, True),
+                                                   (col.LAST_PREGNANCY_DATE, op.le, pop.date
+                                                    - timedelta(days=270))])
+            self.update_sub_pop_test_date(pop, third_trimester_pop, self.prob_anc_test_trim3)
+
+            # get post-delivery population tested during the previous time step
+            post_delivery_pop = pop.get_sub_pop([(col.HIV_STATUS, op.eq, False),
+                                                 (col.LAST_TEST_DATE, op.eq, pop.date - time_step),
+                                                 (col.LAST_PREGNANCY_DATE, op.le, pop.date
+                                                  - (timedelta(days=270) + time_step))])
+            self.update_sub_pop_test_date(pop, post_delivery_pop, self.prob_test_postdel)
```

Comment:
There is a bit of an issue here. This is looking for anyone whose last test date was the previous timestep, and whose last pregnancy was at least (9 months + 1 timestep) ago. But this would apply to _anyone_ who has given birth in the past and is not currently pregnant, even if it were years ago.